examples/wallet/wallet.v.py

```diff
@@ -1,25 +1,46 @@
 # An example of how you can do a wallet in Viper.
 # Warning: NOT AUDITED. Do not use to store substantial quantities of funds.
 
-addrs: address[5]
+# A list of the owners addresses (there are a maximum of 5 owners)
+owners: address[5]
+# The number of owners required to approve a transaction
 threshold: num
+# The number of transactions that have been approved
 seq: num
 
-def __init__(_addrs: address[5], _threshold: num):
+def __init__(_owners: address[5], _threshold: num):
     for i in range(5):
-        if _addrs[i]:
-            self.addrs[i] = _addrs[i]
+        if _owners[i]:
+            self.owners[i] = _owners[i]
     self.threshold = _threshold
 
 def approve(_seq: num, to: address, value: wei_value, data: bytes <= 4096, sigdata: num256[3][5]) -> bytes <= 4096:
+    # Every time the number of approvals starts at 0 (multiple signatures can be added through the sigdata argument)
     approvals = 0
+    # Starts by combining:
+    # 1) The number of transactions approved thus far.
+    # 2) The address the transaction is going to be sent to (can be a contract or a user).
+    # 3) The value in wei that will be sent with this transaction.
+    # 4) The data to be sent with this transaction (usually data is used to deploy contracts or to call functions on contracts, but you can put whatever you want in it).
+    # Takes the sha3 (keccak256) hash of the combination
     h = sha3(concat(as_bytes32(_seq), as_bytes32(to), as_bytes32(value), data))
+    # Then we combine the Ethereum Signed message with our previous hash
+    # Owners will have to sign the below message
     h2 = sha3(concat("\x19Ethereum Signed Message:\n32", h))
+    # Verifies that the caller of approve has entered the correct transaction number
     assert self.seq == _seq
+    # Iterates through all the owners and verifies that there signatures,
+    # given as the sidata argument are correct
     for i in range(5):
         if sigdata[i][0]:
-            assert ecrecover(h2, sigdata[i][0], sigdata[i][1], sigdata[i][2]) == self.addrs[i]
+            # If an invalid signature is given for an owner then the contract throws 
+            assert ecrecover(h2, sigdata[i][0], sigdata[i][1], sigdata[i][2]) == self.owners[i]
+            # For every valid signature increase the number of approvals by 1
             approvals += 1
+    # Throw if the number of approvals is less then the number of approvals required (the threshold)
     assert approvals >= self.threshold
+    # The transaction has been approved
+    # Increase the number of approved transactions by 1
     self.seq += 1
+    # Use raw_call to send the transaction
     return raw_call(to, data, outsize=4096, gas=3000000, value=value)
```

Comment:
`gas` seems randomly chosen. What if I don't send enough to afford that gas cost in conjunction with that value? I like the idea of `approve()` being `@payable`, and `gas` is set to `msg.value - value`. Wouldn't this mean the sender should get their refund? I'm not sure.

Reply:
I agree with you, I'll make `approve()` `@payable` .  I also like the idea of making it so that the `msg.sender` gets any excess `msg.value` leftover after their transaction calls.

Reply:
@DavidKnott is there a method to obtain/return the leftover gas from a `raw_call`?